### PR TITLE
[Feature](bangc-ops): new op roi_pooling_forward

### PR DESCRIPTION
**Pull request by dcmee336**
Thanks for your contribution and we appreciate it a lot. 

## 1. Motivation

Add new op roi_pooling_forward.

## 2. Modification

create bangc-ops/kernels/roi_pooling_forward/roi_pooling_forward.cpp
create bangc-ops//test/mlu_op_gtest/pb_gtest/src/zoo/roi_pooling_forward/roi_pooling_forward.cpp
create bangc-ops//test/mlu_op_gtest/pb_gtest/src/zoo/roi_pooling_forward/roi_pooling_forward.h
create bangc-ops//test/mlu_op_gtest/pb_gtest/src/zoo/roi_pooling_forward/test_case/case_0.prototxt

change bangc-ops/mlu_op.h
change docs/bangc-docs/user_guide/9_operators/index.rst
change bangc-ops/kernels/kernel_wrapper/lib/libextops.a
change bangc-ops/kernels/kernel_wrapper/wrapper.h

## 3. Test Report

If you want to know how to do operator testing, you can see [GTest-User-Guide-zh](https://github.com/Cambricon/mlu-ops/blob/master/docs/GTest-User-Guide-zh.md).

#### 3.1.1 Accuracy Acceptance Standard

For static threshold standard details, see: [MLU-OPS Accuracy Acceptance Standard](https://github.com/Cambricon/mlu-ops/blob/master/docs/MLU-OPS-Accuracy-Acceptance-Standard.md).

- [x] diff1: diff1 <= 3e-3
- [x] diff2: diff2 <= 3e-3

#### 3.1.2 Operator Scheme checklist

|     No.        |                 Details              |            Check Results             |
|----------------|--------------------------------------|--------------------------------------|
|        1       |Supported hardware                    |             MLU370<br>MLU590         |
|        2       |Job types                             |          U1       |
|        3       |Layouts                               |          input, output: NHWC 、rois: ARRAY      |
|        4       |Whether multi-dimensions are supported|                                      |
|        5       |Whether element zero is supported     |                                      |
|        6       |Data type(half/float)                 |           half / float            |
|        7       |Whether there is size limit           |                                      |

#### 3.1.3 New Feature Test

If you have checked the following items, please tick the relevant box.

- [x] Data type test
- [x] Multi-dimensional tensor test
- [x] Layout test
- [x] Different size/integer remainder end segment/alignment misalignment test
- [x] Zero dimensional tensor test/zero element test
- [x] stability test
- [x] Multiple platform test
- [x] Gen_case module test
- [x] Nan/INF tests 
- [ ] Bug fix tests
- [x] For memory leak check details, see[GTest-User-Guide-zh](https://github.com/Cambricon/mlu-ops/blob/master/docs/GTest-User-Guide-zh.md).
- [ ] For code coverage check details, see: [GTest-User-Guide-zh](https://github.com/Cambricon/mlu-ops/blob/master/docs/GTest-User-Guide-zh.md).
- [x] For I/O calculation efficiency check details, see: [MLU-OPS Performance Acceptance Standard](https://github.com/Cambricon/mlu-ops/blob/master/docs/MLU-OPS-Performance-Acceptance-Standard.md).

#### 3.1.4 Parameter Check

When a new operator is submitted, the test points are given and the test results are stated.

|                   Test Point                    | Acceptance Standard | Test Result (Error Message) |
| ----------------------------------------------- | --------------------| --------------------------- |
| Whether it conforms to the operator restriction |     Normal error    |                             |
| Whether illegal parameters are passed           |     Normal error    |                             |

### 3.2 Accuracy Test

For the cases used in the New Feature Test section, the features and the number of cases are recorded here. When multiple operations are tested, multiple tables are needed to include details of these operations.

Operation:

|Test Point           | Description                      | Quantity |  Comment |
|----------           |----------------------------------|----------|  --------|
|Data type test       |half/float/                   | ALL PASS      |          |
|Mult-tensor test     |Supports 1-8 dims                 |NOT SUPPORT          |          |
|Layout test          |Supports NHWC                |ALL PASS          |          |
|Zero element test    |Whether to support this test      |ALL PASS          |          |
|Stability test       |--gtest_repeat=NUM<br>--thread=NUM|ALL PASS          |          |
|mlu-only mode test   |--mlu-only,see [MLU-OPS Performance Acceptance Standard](https://github.com/Cambricon/mlu-ops/blob/master/docs/MLU-OPS-Performance-Acceptance-Standard.md)|ALL PASS          |          |
|Mult-platform test   |MLU370/MLU590                     |ALL PASS          |          |
|Nan/INF test         |Whether to support this test      |ALL PASS          |          |
|Memory leak check    |Test result                       |ALL PASS          |          |
|Code coverage check  |Test result                       |          |          |

### 3.3 Performance Test

See [MLU-OPS Performance Acceptance Standard](https://github.com/Cambricon/mlu-ops/blob/master/docs/MLU-OPS-Performance-Acceptance-Standard.md) for details.

Platform：MLU370

|Operation|Mlu_hardware_time(us)|Mlu_interface_time(us)|Mlu_io_efficiency|Mlu_compute_efficiency|Mlu_workwpace_size(Bytes)|Data_type|Shape|
|-------|----|----|-----|----|----|----|-----|
|roi_pooling_forward|289    |122.2    |0.20961 |0.0007  |0    |float32 |input: [16, 64, 64, 48], output:[10, 12, 12, 68]     |
|roi_pooling_forward|308    |117.069    |0.3593 |0.0003    |0    |float16 |input: [32, 128, 128, 32] output: [16, 12, 12, 32]     |

Platform：MLU590

|Operation|Mlu_hardware_time(us)|Mlu_interface_time(us)|Mlu_io_efficiency|Mlu_compute_efficiency|Mlu_workwpace_size(Bytes)|Data_type|Shape|
|-------|----|----|----|----|----|----|-----|
|roi_pooling_forward|238   |86.6  |0.038    |0.0003    |0    |float32    |input: [16, 64, 64, 48], output:[10, 12, 12, 68]      |
|roi_pooling_forward|371   |93    |0.044    |0.0001    |0    |float16    |input: [32, 128, 128, 32] output: [16, 12, 12, 32]     |

### 3.4 Summary Analysis

Please give a brief overview here, if you want to note and summarize the content.

### CODE REVIEW

File: bangc-ops/mlu_op.h [hunk 1]
```diff
@@ -13711,6 +13727,158 @@ mluOpDiffIouRotatedSortVerticesForward(mluOpHandle_t handle,
...
+ * @param[in] rois
+ * Pointer to the MLU memory that stores the rois tensor. Rois means regions of interest.
+ * @param[in] spatial_scale
+ * The spatial scale of each regions of interest in the input feature map.
+ * @param[in] output_desc
+ * The descriptor of output tensor. For detailed information, see ::mluOpTensorDescriptor_t.
+ * @param[out] output
+ * Pointer to the MLU memory that stores the output tensor.
+ * @param[out] argmax
+ * Pointer to the MLU memory that stores the argmax tensor. Pointer may be NULL. Argmax tensor
```
Comment (AquamarineSu) [inline bangc-ops/mlu_op.h:13756]:
```suggestion
 * Pointer to the MLU memory that stores the argmax tensor. This pointer may be NULL. The tensor \b argmax
```

File: bangc-ops/mlu_op.h [hunk 2]
```diff
@@ -13711,6 +13727,158 @@ mluOpDiffIouRotatedSortVerticesForward(mluOpHandle_t handle,
...
+ * means input feature index of maximum for each Roi(Regions of Interest).
+ *
+ * @par Return
+ * - ::MLUOP_STATUS_SUCCESS, ::MLUOP_STATUS_BAD_PARAM.
+ *
+ * @par Data Type
+ * - This function supports the following data types for input tensor \b input, rois tensor \b rois,
+ *   output tensor \b output, and argmax tensor \b argmax. The data type of the \b input \b rois \b output
```
Comment (AquamarineSu) [inline bangc-ops/mlu_op.h:13764]:
```suggestion
 *   output tensor \b output, and argmax tensor \b argmax. The data type of \b input, \b rois, and \b output
```

File: bangc-ops/mlu_op.h [hunk 3]
```diff
@@ -13711,6 +13727,158 @@ mluOpDiffIouRotatedSortVerticesForward(mluOpHandle_t handle,
...
+ *   should be the same. The data type of the argmax is different from others.
```
Comment (AquamarineSu) [inline bangc-ops/mlu_op.h:13765]: The data type of the argmax is different from others. 感觉这句可以删掉呀
```suggestion
 *   must be the same. The data type of the argmax is different from others.
```

File: bangc-ops/mlu_op.h [hunk 4]
```diff
@@ -13711,6 +13727,158 @@ mluOpDiffIouRotatedSortVerticesForward(mluOpHandle_t handle,
...
+ *   - input tensor: half, float.
+ *   - rois tensor: half, float.
+ *   - output tensor: half, float.
+ *   - argmax tensor: int32.
+ *
+ * @par Data Layout
+ * - The support data layout of \b input, \b rois, \b output, \b argmax are as follows:
```
Comment (AquamarineSu) [inline bangc-ops/mlu_op.h:13772]:
```suggestion
 * - The supported data layout of \b input, \b rois, \b output, and \b argmax are as follows:
```

File: bangc-ops/mlu_op.h [hunk 5]
```diff
@@ -13711,6 +13727,158 @@ mluOpDiffIouRotatedSortVerticesForward(mluOpHandle_t handle,
...
+ *   - input tensor: \b MLUOP_LAYOUT_NHWC.
+ *   - rois tensor: \b MLUOP_LAYOUT_ARRAY, only supports 2D tensor.
```
Comment (AquamarineSu) [inline bangc-ops/mlu_op.h:13774]:
```suggestion
 *   - ROIs tensor: \b MLUOP_LAYOUT_ARRAY (only supports 2D tensor).
```

File: bangc-ops/mlu_op.h [hunk 6]
```diff
@@ -13711,6 +13727,158 @@ mluOpDiffIouRotatedSortVerticesForward(mluOpHandle_t handle,
...
+ *   - output tensor: \b MLUOP_LAYOUT_NHWC.
+ *   - argmax tensor: \b MLUOP_LAYOUT_NHWC.
+ *
+ * @par Scale Limitation
+ * - The value of \b pooling_mode only support \p MLUOP_POOLING_MAX.
```
Comment (AquamarineSu) [inline bangc-ops/mlu_op.h:13779]:
```suggestion
 * - The value of \b pooling_mode only supports \p MLUOP_POOLING_MAX.
```

File: bangc-ops/mlu_op.h [hunk 7]
```diff
@@ -13711,6 +13727,158 @@ mluOpDiffIouRotatedSortVerticesForward(mluOpHandle_t handle,
...
+ * - The input tensor, output tensor and argmax tensor must have four dimensions.
+ * - Data type of half of \b input and \b rois is not recommended due to low precision.
```
Comment (AquamarineSu) [inline bangc-ops/mlu_op.h:13781]: 这句没太看懂

File: bangc-ops/mlu_op.h [hunk 8]
```diff
@@ -13711,6 +13727,158 @@ mluOpDiffIouRotatedSortVerticesForward(mluOpHandle_t handle,
...
+ * - Size of the lowest dimension of input tensors, output tensor and argmax tensor must be the same.
```
Comment (AquamarineSu) [inline bangc-ops/mlu_op.h:13782]:
```suggestion
 * - The size of the lowest dimension of input tensors, output tensor, and argmax tensor must be the same.
```

File: bangc-ops/mlu_op.h [hunk 9]
```diff
@@ -13711,6 +13727,158 @@ mluOpDiffIouRotatedSortVerticesForward(mluOpHandle_t handle,
...
+ * - The total number of dimension of rois tensor must be 2.
```
Comment (AquamarineSu) [inline bangc-ops/mlu_op.h:13783]:
```suggestion
 * - The total number of dimensions of ROIs tensor must be 2.
```

File: bangc-ops/mlu_op.h [hunk 10]
```diff
@@ -13711,6 +13727,158 @@ mluOpDiffIouRotatedSortVerticesForward(mluOpHandle_t handle,
...
+ * - Size of the highest dimension of output tensor and rois tensor must be the same.
```
Comment (AquamarineSu) [inline bangc-ops/mlu_op.h:13784]:
```suggestion
 * - The size of the highest dimension of output tensor and the ROIs tensor must be the same.
```

File: bangc-ops/mlu_op.h [hunk 11]
```diff
@@ -13711,6 +13727,158 @@ mluOpDiffIouRotatedSortVerticesForward(mluOpHandle_t handle,
...
+ * - The shape of \b rois should be [rois_num, 5]. \b rois_num means the total number of \b rois.
```
Comment (AquamarineSu) [inline bangc-ops/mlu_op.h:13785]:
```suggestion
 * - The shape of \b rois must be [rois_num, 5]. \b rois_num means the total number of \b rois.
```

File: bangc-ops/mlu_op.h [hunk 12]
```diff
@@ -13711,6 +13727,158 @@ mluOpDiffIouRotatedSortVerticesForward(mluOpHandle_t handle,
...
+ * the nram_limitation is (163804 - 4 * \b channels) / 2. \b pooled_h means height of output. \b pooled_w means width of
+ * output.
+ *
+ * @par API Dependency
+ * - None
+ *
+ * @par Performance Optimization
+ * - None
+ *
+ * @node
```
Comment (AquamarineSu) [inline bangc-ops/mlu_op.h:13805]:
```suggestion
 * @note
```

File: bangc-ops/mlu_op.h [hunk 13]
```diff
@@ -13711,6 +13727,158 @@ mluOpDiffIouRotatedSortVerticesForward(mluOpHandle_t handle,
...
+ * - It is not recommended to set the data type of grads tensor, rois tensor and grads_image tensors
```
Comment (AquamarineSu) [inline bangc-ops/mlu_op.h:13806]: grads tensor指什么？

File: bangc-ops/mlu_op.h [hunk 14]
```diff
@@ -13711,6 +13727,158 @@ mluOpDiffIouRotatedSortVerticesForward(mluOpHandle_t handle,
...
+ *   that may cause the low accuracy on MLU200 series.
+ * - When the input data or parameter contains NaN or infinity:
+ *   - On MLU200 series:
+ *     - The \b output is the positive saturation value. The \b argmax is the index of the last NaN in the kernel of the
```
Comment (AquamarineSu) [inline bangc-ops/mlu_op.h:13810]: 从排版的角度，不建议用二级列表，和上面200系列合成一句话吧

File: bangc-ops/mlu_op.h [hunk 15]
```diff
@@ -13711,6 +13727,158 @@ mluOpDiffIouRotatedSortVerticesForward(mluOpHandle_t handle,
...
+ *       pooling.
+ *   - On MLU300 series and CE3226:
```
Comment (AquamarineSu) [inline bangc-ops/mlu_op.h:13812]: 这个型号是不是不能对外写？

File: bangc-ops/mlu_op.h [hunk 16]
```diff
@@ -13711,6 +13727,158 @@ mluOpDiffIouRotatedSortVerticesForward(mluOpHandle_t handle,
...
+ *     - If the last value in the kernel of the pooling is NaN, the \b argmax is the index of the last value, the \b
```
Comment (AquamarineSu) [inline bangc-ops/mlu_op.h:13813]:
```suggestion
 *     - If the last value in the kernel of the pooling is NaN, \b argmax is the index of the last value,  \b
```

File: bangc-ops/mlu_op.h (hunk 7 above)
Comment (dcmee336) [reply bangc-ops/mlu_op.h:13781]: 就是由于half类型精度低所以不推荐使用，开发者写的一句废话好像，已删除。

File: bangc-ops/mlu_op.h (hunk 13 above)
Comment (dcmee336) [reply bangc-ops/mlu_op.h]: 应该是input tensor和output tensor，已修改。

**Comment by duzekunKTH** [comment]
commit 改为 `[Feature](bangc-ops): new op roi_pooling_forward ` 吧。
同时 CI 挂了请看一下。

File: bangc-ops/mlu_op.h [hunk 17]
```diff
@@ -13711,6 +13727,155 @@ mluOpDiffIouRotatedSortVerticesForward(mluOpHandle_t handle,
                                        const mluOpTensorDescriptor_t idx_desc,
                                        void *idx);
 
+// Group:RoiPoolingForward
+/*!
+ * @brief Generates a fixed size feature map and input feature index
+ * of argmax for each Roi (Regions of Interest) to perform ::mluOpRoiPoolingForward operation.
```
Comment (AquamarineSu) [inline bangc-ops/mlu_op.h:13733]:
```suggestion
 * of argmax for each ROI (Regions of Interest) to perform ::mluOpRoiPoolingForward operation.
```

File: bangc-ops/mlu_op.h [hunk 18]
```diff
@@ -13711,6 +13727,155 @@ mluOpDiffIouRotatedSortVerticesForward(mluOpHandle_t handle,
...
+ * The pooling mode of ROI Pooling Forward defined in ::mluOpPoolingMode_t.
+ * @param[in] input_desc
+ * The descriptor of the input tensor in the roipoolingforward process. For detailed
+ * information, see ::mluOpTensorDescriptor_t.
+ * @param[in] input
+ * Pointer to the MLU memory that stores the input tensor.
+ * @param[in] rois_desc
+ * The descriptor of the ROIs tensor. For detailed information, see ::mluOpTensorDescriptor_t.
+ * @param[in] rois
+ * Pointer to the MLU memory that stores the rois tensor.
```
Comment (AquamarineSu) [inline bangc-ops/mlu_op.h:13748]:
```suggestion
 * Pointer to the MLU memory that stores the ROIs tensor.
```

File: bangc-ops/test/mlu_op_gtest/pb_gtest/src/zoo/roi_pooling_forward/roi_pooling_forward.cpp [hunk 19]
```diff
@@ -0,0 +1,157 @@
...
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *************************************************************************/
+#include <string>
+#include <algorithm>
+#include "roi_pooling_forward.h"
+#define getParam(ty, ctx) \
+  (ty) parser_->getProtoNode()->roi_pooling_forward_param().ctx()
+#define getTensorDims(x, y) tensor_desc_[x].tensor->dims[y]
+#define getTensorDesc(x) tensor_desc_[x].tensor
+#define getDevicePtr(x) data_vector_[x].device_ptr
```
Comment (PetrelYy) [inline bangc-ops/test/mlu_op_gtest/pb_gtest/src/zoo/roi_pooling_forward/roi_pooling_forward.cpp:30]:
```suggestion
#include "roi_pooling_forward.h"

#include <algorithm>
#include <string>

#define getParam(ty, ctx) \
  (ty) parser_->getProtoNode()->roi_pooling_forward_param().ctx()
#define getTensorDims(x, y) tensor_desc_[x].tensor->dims[y]
#define getTensorDesc(x) tensor_desc_[x].tensor
#define getDevicePtr(x) data_vector_[x].device_ptr
```

File: bangc-ops/test/mlu_op_gtest/pb_gtest/src/zoo/roi_pooling_forward/roi_pooling_forward.h [hunk 20]
```diff
@@ -0,0 +1,74 @@
...
+                            float *argmax);
+
+ private:
+  int batch;
+  int height;
+  int width;
+  int channels;
+  int rois_num;
+  int pool_height;
+  int pool_width;
```
Comment (PetrelYy) [inline bangc-ops/test/mlu_op_gtest/pb_gtest/src/zoo/roi_pooling_forward/roi_pooling_forward.h:59]:
```suggestion
  int batch_;
  int height_;
  int width_;
  int channels_;
  int rois_num_;
  int pool_height_;
  int pool_width_;
```

File: bangc-ops/mlu_op.h [hunk 21]
```diff
@@ -13711,6 +13727,155 @@ mluOpDiffIouRotatedSortVerticesForward(mluOpHandle_t handle,
...
+ * - This function supports the following data types for input tensor \b input, rois tensor \b rois,
+ *   output tensor \b output, and argmax tensor \b argmax. The data type of \b input, \b rois, and \b output
+ *   must be the same.
+ *   - input tensor: half, float.
+ *   - rois tensor: half, float.
+ *   - output tensor: half, float.
+ *   - argmax tensor: int32.
+ *
+ * @par Data Layout
+ * - The supported data layout of \b input, \b rois, \b output, \b argmax are as follows:
```
Comment (AquamarineSu) [inline bangc-ops/mlu_op.h:13772]:
```suggestion
 * - The supported data layouts for \b input, \b rois, \b output, and \b argmax are as follows:
```

File: bangc-ops/mlu_op.h [hunk 22]
```diff
@@ -13711,6 +13727,155 @@ mluOpDiffIouRotatedSortVerticesForward(mluOpHandle_t handle,
...
+ *   - input tensor: \b MLUOP_LAYOUT_NHWC.
+ *   - ROIs tensor: \b MLUOP_LAYOUT_ARRAY, only supports 2D tensor.
+ *   - output tensor: \b MLUOP_LAYOUT_NHWC.
+ *   - argmax tensor: \b MLUOP_LAYOUT_NHWC.
+ *
+ * @par Scale Limitation
+ * - The value of \b pooling_mode only supports \p MLUOP_POOLING_MAX.
+ * - The input tensor, output tensor and argmax tensor must have four dimensions.
```
Comment (AquamarineSu) [inline bangc-ops/mlu_op.h:13780]:
```suggestion
 * - The input tensor, output tensor, and argmax tensor must have four dimensions.
```

File: bangc-ops/mlu_op.h [hunk 23]
```diff
@@ -13711,6 +13727,155 @@ mluOpDiffIouRotatedSortVerticesForward(mluOpHandle_t handle,
...
+ * - The size of the lowest dimension of input tensors, output tensor and argmax tensor must be the same.
```
Comment (AquamarineSu) [inline bangc-ops/mlu_op.h:13781]:
```suggestion
 * - The size of the lowest dimension of input tensors, output tensor, and argmax tensor must be the same.
```

File: bangc-ops/mlu_op.h [hunk 24]
```diff
@@ -13711,6 +13727,155 @@ mluOpDiffIouRotatedSortVerticesForward(mluOpHandle_t handle,
...
+ * - The total number of dimensions of ROIs tensor must be 2.
+ * - The size of the highest dimension of output tensor and rois tensor must be the same.
+ * - The shape of \b rois must be [rois_num, 5]. \b rois_num means the total number of \b rois.
+ * - \b Rois[i] consists of [batch_id, x1, y1, x2, y2]. \b batch_id should be in the range of
+ * [0, batch_num - 1]. \b x1 and \b y1 should be greater than or equal to 0. \b x2 should be
+ * greater than \b x1. \b y2 should be greater than \b y1. \b batch_id means id of batch of \b rois. \b x1, \b y1,
```
Comment (AquamarineSu) [inline bangc-ops/mlu_op.h:13787]: 建议把 should 批量替换成must，means 换成represents，
```suggestion
 * greater than \b x1. \b y2 must be greater than \b y1. \b batch_id represents ID of the batch of \b rois. \b x1, \b y1,
```

File: bangc-ops/mlu_op.h [hunk 25]
```diff
@@ -13711,6 +13727,155 @@ mluOpDiffIouRotatedSortVerticesForward(mluOpHandle_t handle,
...
+ * output.
+ *
+ * @par API Dependency
+ * - None
+ *
+ * @par Performance Optimization
+ * - None
+ *
+ * @note
+ * - It is not recommended to set the data type of input tensor, rois tensor and output tensors
```
Comment (AquamarineSu) [inline bangc-ops/mlu_op.h:13805]:
```suggestion
 * - It is not recommended to set the data type of input tensor, ROIs tensor, and output tensors
```

File: bangc-ops/mlu_op.h [hunk 26]
```diff
@@ -13711,6 +13727,155 @@ mluOpDiffIouRotatedSortVerticesForward(mluOpHandle_t handle,
...
+ *   that may cause the low accuracy on MLU200 series.
+ * - When the input data or parameter contains NaN or infinity:
+ *   - On MLU200 series, the \b output is the positive saturation value.
+ *     The \b argmax is the index of the last NaN in the kernel of the pooling.
+ *   - On MLU300 series, If the last value in the kernel of the pooling is NaN, \b argmax is
```
Comment (AquamarineSu) [inline bangc-ops/mlu_op.h:13810]:
```suggestion
 *   - On MLU300 series, if the last value in the kernel of the pooling is NaN, \b argmax is
```

**Review by AquamarineSu — APPROVED** [review]

File: bangc-ops/test/mlu_op_gtest/pb_gtest/src/zoo/roi_pooling_forward/roi_pooling_forward.cpp (hunk 19 above)
Comment (dcmee336) [reply bangc-ops/test/mlu_op_gtest/pb_gtest/src/zoo/roi_pooling_forward/roi_pooling_forward.cpp]: 已修改

File: bangc-ops/test/mlu_op_gtest/pb_gtest/src/zoo/roi_pooling_forward/roi_pooling_forward.h (hunk 20 above)
Comment (dcmee336) [reply bangc-ops/test/mlu_op_gtest/pb_gtest/src/zoo/roi_pooling_forward/roi_pooling_forward.h]: 已修改

**Review by Weijie-itsum — APPROVED** [review]

**Review by PetrelYy — APPROVED** [review]

**Pull request merged**